consensus/ethash/ethash.go

```diff
@@ -561,14 +629,40 @@ func (ethash *Ethash) SetThreads(threads int) {
 
 // Hashrate implements PoW, returning the measured rate of the search invocations
 // per second over the last minute.
+// Note the returned hashrate includes local hashrate, but also includes the total
+// hashrate of all remote miner.
 func (ethash *Ethash) Hashrate() float64 {
-	return ethash.hashrate.Rate1()
+	var (
+		total uint64
+		errCh = make(chan error)
+	)
+	// getHashrate gathers all remote miner's hashrate.
+	getHashrate := func(hashrate map[common.Hash]hashrate) {
+		for _, rate := range hashrate {
+			// this could overflow
+			total += rate.rate
+		}
+		return
+	}
+	op := &remoteOp{
+		typ:    opHashrate,
+		rateOp: getHashrate,
+		errCh:  errCh,
+	}
+	ethash.remoteOp <- op
+	<-errCh
+
+	return ethash.hashrate.Rate1() + float64(total)
 }
 
-// APIs implements consensus.Engine, returning the user facing RPC APIs. Currently
-// that is empty.
+// APIs implements consensus.Engine, returning the user facing RPC APIs.
 func (ethash *Ethash) APIs(chain consensus.ChainReader) []rpc.API {
-	return nil
+	return []rpc.API{{
+		Namespace: "eth",
+		Version:   "1.0",
+		Service:   &API{ethash},
+		Public:    true,
+	}}
```

Comment:
Just to make it nicer for future APIs, lets also advertize these in the `ethash` namespace, similar to how `clique` is done.